lib/rdkafka/producer.rb

```diff
@@ -295,6 +295,31 @@ def arity(callback)
       callback.method(:call).arity
     end
 
+    # Set the OAuthBearer token
+    #
+    # @param token [String] The token value
+    # @param lifetime_ms [Integer] The token lifetime in milliseconds since the epoch
+    # @param principal_name [String] The principal name
+    # @param extensions [String] The token extensions
+    # @param extension_size [Integer] The token extensions size
+    # @return [nil]
+    # @raise [Rdkafka::RdkafkaError] when setting the token fails
+    def oauthbearer_set_token(token, lifetime_ms, principal_name, extensions, extension_size)
```

Comment:
@mensfeld I think you mean something like this, with a corresponding function in consumer and admin.  I think we only have to have wrap the `rdkafka_oauthbearer_set_token` binding in Ruby as it can handle calling `rdkafka_oauthbearer_set_token_failure` based on the response.

Reply:
Yeah I somehow missed that as I was checking the sources - it's missing from consumer hence your PR will not allow for consumer nor admin operations. Yes this is what we aim to do :)